website/content/docs/concepts/oidc-provider.mdx

```diff
@@ -0,0 +1,144 @@
+
+The endpoint will [validate](https://openid.net/specs/openid-connect-core-1_0.html#AuthRequestValidation) client requests ensuring all required parameters are present and valid. The `redirect_uri` of the request will be validated against the client's `redirect_uris`. The requesting Vault entity will be validated against the client's `assignments`. An appropriate [error code](https://openid.net/specs/openid-connect-core-1_0.html#AuthError) is returned for invalid requests.
+
+An authorization code is generated with a successful validation of the request. The authorization code is single-use and cached with a lifetime of approximately 5 minutes, which mitigates the risk of leaks. A response including the original `state` presented by the client and `code` will be returned to the Vault UI and initiates the request. Vault will issue an HTTP 302 redirect to the `redirect_uri` of the request, which includes the `code`, `state`, and `nonce` as query parameters.
+
+### Token Endpoint
+
+Each provider will offer a [token endpoint](https://openid.net/specs/openid-connect-core-1_0.html#TokenEndpoint). The endpoint may be unauthenticated in Vault but is authenticated by requiring a `client_secret` as described in [client authentication](https://openid.net/specs/openid-connect-core-1_0.html#ClientAuthentication). The endpoint ingests all required [token request](https://openid.net/specs/openid-connect-core-1_0.html#TokenRequest) parameters as input. The endpoint [validates](https://openid.net/specs/openid-connect-core-1_0.html#TokenRequestValidation) the client requests and exchanges an authorization code for the ID token and access token. The cache of authorization codes will be verified against the code presented in the exchange. The appropriate [error code](https://openid.net/specs/openid-connect-core-1_0.html#TokenErrorResponse) are returned for all invalid requests.
+
+The ID token is generated and returned upon successful client authentication and request validation. The ID token will contain a combination of required and configurable claims. The required claims are enumerated in the scopes section above for the `openid` scope. The configurable claims are populated by templates associated with the scopes provided in the authentication request that generated the authorization code.
+
+An access token is also generated and returned upon successful client authentication and request validation. The access token is a Vault [batch token](https://www.vaultproject.io/docs/concepts/tokens#batch-tokens) with a policy that only provides read access to the issuing provider's [userinfo endpoint](https://openid.net/specs/openid-connect-core-1_0.html#UserInfo). The access token is also a TTL as defined by the `access_token_ttl` of the requesting client.
```

Comment:
Maybe link to vault API docs for the userinfo endpoint: https://github.com/hashicorp/vault/blame/main/website/content/api-docs/secret/identity/oidc-provider.mdx#L705

Reply:
added the preliminary links but they won't be hot until we release 1.9.